tests/unit/swagger/schemaDetails3.spec.ts

```diff
@@ -1156,6 +1157,31 @@ describe('Definition generation for OpenAPI 3.0.0', () => {
                     ],
                   },
                   id: { type: 'number', format: 'double', default: undefined, description: undefined, example: undefined },
+                  indexedResponse: {
+                    default: undefined,
+                    description: undefined,
+                    example: undefined,
+                    format: undefined,
+                    properties: {
+                      data: {
+                        $ref: '#/components/schemas/Record',
```

Comment:
This does not seem right.

Reply:
This is what I was talking about, and it's not related to this PR, it's just something I've noticed trying to debug my projects where I have tsoa (which was crashing without this PR)

Reply:
It's good that you found this issue, but I think the change you are proposing here is trying to recover from a situation we should prevent earlier on.
As you noted, not having the referencer here is the red flag: It looks like we are not correctly propagating that. If you try to reconstruct the type afterwards without the referencer, we are missing major context we need, especially the type arguments.

Reply:
Hm right, so what do you think we need to do here?

Reply:
If you want to debug, try to step through from the point where the TypeReference node is resolved until the point where the referencer is gone.Somewhere in between is the issue :laughing:  Hope the "Pretest"debug config comes in handy. 
Wish I could tell you what the issue is, but I can't tell until I get time to debug this myself.

Reply:
I've updated test model with a version of type more simple which still cause this issue.

I've tried to inspect, but I haven't found anything, the only things I have noticed are that [declaration](https://github.com/lukeautry/tsoa/pull/676/files#diff-04ed568b4a647b0bfd718a5afdb822fbR262) have `aliasTypeArguments` and `aliasSymbol` properties which seems to be the final type we need (`Record<'id', string>`) and when we use `typeToTypeNode` on this declaration ([here](https://github.com/lukeautry/tsoa/pull/676/files#diff-04ed568b4a647b0bfd718a5afdb822fbR264)) we have an `anonymous` node where `pos` aren't defined (which I think cause the issue).
And then, when we are in the [mapped type scope](https://github.com/lukeautry/tsoa/pull/676/files#diff-04ed568b4a647b0bfd718a5afdb822fbR264), using `getTypeFromTypeNode` on `this.referencer` doesn't work, indeed it create an error node. I've tried to use `getTypeFromTypeNode` on `this.typeNode` and we get back a type with `aliasTypeArguments` and `aliasSymbol`, and using `type.getProperties()` doesn't returns any props.